pkg/network/base_connection.go

```diff
@@ -154,45 +173,20 @@ func (t *baseTcpConnection) consumeConnection(conn net.Conn) {
 	wg.Wait()
 
 	t.logger.Debugf("Stopped consuming connection with local %s and remote %s", conn.LocalAddr().String(), conn.RemoteAddr().String())
-	t.connMutex.RLock()
-	err := t.conn.Close()
-	t.connMutex.RUnlock()
-	if err != nil && !isEOF(err) && !isUseOfClosedConnection(err) {
-		t.logger.Warnf("Error while closing the previous connection: %s", err)
-	}
 }
 
-func (t *baseTcpConnection) tryPropagateError(err error) {
-	select {
-	case <-t.ctx.Done():
-		return
-	default:
-		t.errors <- err
-	}
+func (t *baseTcpConnection) pushError(err error) {
+	t.errors <- err
 }
 
-func (t *baseTcpConnection) tryPushOutboundChannel(msg *ctrl.OutboundMessage) {
-	select {
-	case <-t.ctx.Done():
-		return
-	default:
-		t.outboundMessageChannel <- msg
-	}
+func (t *baseTcpConnection) pushOutboundChannel(msg *ctrl.OutboundMessage) {
+	t.outboundMessageChannel <- msg
 }
 
-func (t *baseTcpConnection) close() (err error) {
-	t.connMutex.RLock()
-	if t.conn != nil {
-		err = t.conn.Close()
-	}
-	t.connMutex.RUnlock()
+func (t *baseTcpConnection) close() {
```

Comment:
Why not move this cleanup up into the cleanup goroutine to avoid any possible race?